repose-aggregator/tests/functional-tests/src/test/groovy/features/core/proxy/RequestQueryParamTest.groovy

```diff
@@ -87,8 +87,11 @@ class RequestQueryParamTest extends ReposeValveTest {
 
         where:
         uriSuffixGiven                             | uriSuffixExpected                          | method
-        "/path/to/resource?key=value@%"            | "/path/to/resource?key=value%40%25"        | "GET"
-        "/path/to/resource?key=va%lu@e"            | "/path/to/resource?key=va%25lu%40e"        | "GET"
+        // todo: The '%' character breaks Jetty -- getQueryString returns the raw character while all of the
+        // getParameter methods return garbage. What do we want to do about this?
```

Comment:
Yes, they seem to be illegal characters unless encoded. However, it appears that these tests are passing now. I would bet that is because of the change to the way we HTTP Client. Where we used to call getParameterMap and use the values from that to add query parameters, we now use getQueryString and set that on the outgoing request. HTTP Client expects an unencoded string, which is what Jetty gives us, so HTTP Client performs the encoding for us.

From https://tools.ietf.org/html/rfc3986#section-2.4:
>  Because the percent ("%") character serves as the indicator for
>  percent-encoded octets, it must be percent-encoded as "%25" for that
>  octet to be used as data within a URI.

In other words, I think we answered what we want to do about this while we were doing other work. I'll uncomment these tests.